api/internal/testutil/server.go

```diff
@@ -296,21 +296,29 @@ func (s *TestServer) waitForAPI() {
 	)
 }
 
-// waitForLeader waits for the Nomad server's HTTP API to become
-// available, and then waits for a known leader and an index of
-// 1 or more to be observed to confirm leader election is done.
-func (s *TestServer) waitForLeader() {
+// waitForServers waits for the Nomad server's HTTP API to become available,
+// and then waits for the keyring to be intialized. This implies a leader has
+// been elected and Raft writes have occurred.
+func (s *TestServer) waitForServers() {
```

Comment:
For this and the corresponding change in `testutil/server.go` I decided to switch all tests to wait for the keyring to be initialized.

I did this because I think these tests are more like integration tests and less concerned with the implementation details than `nomad/` package tests. Therefore I thought it appropriate to let these tests block until servers are *ready* by default.